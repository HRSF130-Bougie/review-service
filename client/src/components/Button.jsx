import React from 'react';
import styled from 'styled-components';


const Show = styled.button`
background-color: rgb(255, 255, 255) ;
border-radius: 8px ;
border-width: 1px ;
border-style: solid ;
padding: 13px 23px ;
cursor: pointer;
width: auto ;
font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, "Helvetica Neue", sans-serif !important;
font-size: 16px !important;
    line-height: 20px !important;
    font-weight: 600 !important;
    : hover {
        background-color:#f9f9f9;
    }
`;



const Button = (props) => {
return(
    <div>
        <Show>{`Show all ${props.reviewsNum} reviews`}</Show>
        <dialog open>This is an open dialog window</dialog>
    </div>
)
}

export default Button;