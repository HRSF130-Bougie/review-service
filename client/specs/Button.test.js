import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/Button.jsx';

describe('My compnent', () => {
  const testProps = [
    {
      ID: 244,
      visitorName: 'Ms. Lula Shanahan',
      avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/langate/128.jpg',
      review: 'Expedita non est asperiores voluptatem eius sit. Sapiente qui debitis sint blanditiis cum hic. Tempora iure nam.',
      hotelID: 57,
      col: 'August 2018',
      id: 57,
      cleanRate: 2.6,
      commRate: 4,
      checkinRate: 3.8,
      acuurRate: 3.3,
      locRate: 3.9,
      valueRate: 4.1,
    },
  ];

  it('it works', () => {
    const tree = renderer.create(<Button reviews={testProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
