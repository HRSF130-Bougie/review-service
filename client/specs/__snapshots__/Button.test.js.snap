// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My compnent it works 1`] = `
<div>
  <button
    className="sc-iwaifL emuFVf"
    onClick={[Function]}
  >
    Show all 1 reviews
  </button>
</div>
`;
